addons/account/models/res_partner_bank.py

```diff
@@ -281,14 +281,20 @@ def write(self, vals):
         # leaves them vulnerable to edits via the shell/... So we need to ensure that the user has the rights to edit
         # these fields when writing too.
         # While we do lock changes if the account is trusted, we still want to allow to change them if we go from not trusted -> trusted or from trusted -> not trusted.
-        any_trusted_accounts = any(account.lock_trust_fields for account in self)
+        trusted_accounts = self.filtered(lambda x: x.lock_trust_fields)
+        any_trusted_accounts = bool(trusted_accounts)
         if not any_trusted_accounts:
             should_allow_changes = True  # If we were on a non-trusted account, we will allow to change (setting/... one last time before trusting)
         else:
             # If we were on a trusted account, we only allow changes if the account is moving to untrusted.
             should_allow_changes = ('allow_out_payment' in vals and vals['allow_out_payment'] is False)
 
-        if ('acc_number' in vals or 'partner_id' in vals) and not should_allow_changes:
+        lock_fields = {'acc_number', 'sanitized_acc_number', 'partner_id', 'acc_type'}
+        if not should_allow_changes and any(
+            account[fname] != account._fields[fname].convert_to_record(vals[fname], account)
```

Comment:
there is no bug for now but sementically it is not so correct.
the input of `convert_to_record` should be cache format instead of write format
could you try 
`account[fname] != account.new(fname: vals[fname])[fname]`
or `account[fname] != account.new(fname: account._fields[fname].convert_to_write(vals[fname], account))[fname]`
if doesn't work at least `convert_to_cache` first

Reply:
Nice, didn't think of this one